flagsmith/resource_flag.go

```diff
@@ -0,0 +1,233 @@
+			},
+			"feature": {
+				MarkdownDescription: "ID of the feature",
+				Required:            true,
+				Type:                types.NumberType,
+			},
+			"environment": {
+				MarkdownDescription: "ID of the environment",
+				Required:            true,
+				Type:                types.NumberType,
+			},
+		},
+	}, nil
+}
+
+type flagResource struct {
+	provider provider
+}
+
+func (t flagResourceType) NewResource(ctx context.Context, in tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	provider, diags := convertProviderType(in)
+
+	return flagResource{
+		provider: provider,
+	}, diags
+}
+
+func (r flagResource) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	var data FlagResourceData
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Read and load the state of the object
+	readResponse := tfsdk.ReadResourceResponse{State: resp.State}
+	r.Read(ctx, tfsdk.ReadResourceRequest{
+		State: tfsdk.State{
+			Raw:    req.Plan.Raw,
+			Schema: req.Plan.Schema,
+		},
+		ProviderMeta: req.ProviderMeta,
+	}, &readResponse)
+	if readResponse.Diagnostics.HasError() {
+		resp.Diagnostics.Append(readResponse.Diagnostics...)
+		tflog.Error(ctx, "Error reading resource state")
+		return
+	}
+	// Log the state
+	elog := fmt.Sprintf("%+v", readResponse.State.Get(ctx, &FlagResourceData{}))
+
+	tflog.Debug(ctx, elog)
+
+	//Now call update to update the state
+	updateResponse := tfsdk.UpdateResourceResponse{State: resp.State}
+	r.Update(ctx, tfsdk.UpdateResourceRequest{
+		Config:       req.Config,
+		Plan:         req.Plan,
+		State:        readResponse.State,
+		ProviderMeta: req.ProviderMeta,
+	}, &updateResponse)
+	if updateResponse.Diagnostics.HasError() {
+		resp.Diagnostics.Append(updateResponse.Diagnostics...)
+		tflog.Error(ctx, "Error updating resource state")
+		return
+	}
+	resp.State = updateResponse.State
+	resp.Diagnostics.Append(updateResponse.Diagnostics...)
+
+}
+func (r flagResource) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	var data FlagResourceData
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	// Early return if the state is wrong
+	if diags.HasError() {
+		return
+	}
+	featureState, err := r.provider.client.GetEnvironmentFeatureState(data.EnvironmentKey.Value, data.FeatureName.Value)
+	if err != nil {
+		panic(err)
+	}
+	resoureData := MakeFlagResourceDataFromClientFS(featureState)
+	resoureData.EnvironmentKey = data.EnvironmentKey
+	resoureData.FeatureName = data.FeatureName
+	elog := fmt.Sprintf("%+v", resoureData.FeatureStateValue)
+	tflog.Debug(ctx, elog)
+	diags = resp.State.Set(ctx, &resoureData)
+	if diags.HasError() {
+		// Log error from diags
+		for _, diag := range diags {
+			tflog.Error(ctx, diag.Detail())
+		}
+		resp.Diagnostics.Append(diags...)
+		tflog.Error(ctx, "Read: Error setting resource state")
```

Comment:
Not sure why the error message here is `"Read: Error setting resource state"` ? 'Read' seems to be irrelevant here?

Reply:
It's a common pattern in go log the error with function/mod name